logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -194,7 +194,12 @@ class ConversationEventReceiverImpl(
 
     private suspend fun handleMemberJoin(event: Event.Conversation.MemberJoin) =
         // Attempt to fetch conversation details if needed, as this might be an unknown conversation
-        conversationRepository.fetchConversationIfUnknown(event.conversationId).run {
+        conversationRepository.fetchConversationIfUnknown(event.conversationId)
+            .flatMap {
+                // fetch required unknown users that haven't been persisted during slow sync, e.g. from another team
+                userRepository.fetchUsersIfUnknownByIds(event.members.map { it.id }.toSet())
```

Comment:
Instead of duplicating this line of code, how about creating a new internal UseCase `AddMemberToConversation` that does check if the member info is stored locally and fetch it otherwise, this can help to reduce the complexity of some functions and make inserting members into conversations less prone to errors (we always make sure the user info is in the DB)

Reply:
There is already usecase for adding members to conversation so I just made sure it executes `persistMembers` and made this method do what's needed in order to persist it properly, so there won't be any other methods needed to do it